### PR TITLE
EnableBehavior defaults to ALL if not specified

### PR DESCRIPTION
**Pull request by vishnuravi**
# EnableBehavior defaults to ALL if not specified

## :recycle: Current situation & Problem
If a QuestionnaireItem has > 1 `enableWhen` expressions, but does not have `enableBehavior` specified, the `enableWhen` expressions are currently not applied. Since `enableBehavior` is [not a mandatory element](https://build.fhir.org/ig/HL7/sdc/StructureDefinition-sdc-questionnaire-definitions.html#Questionnaire.item.enableBehavior) in the HL7 SDC it may not be present and a default behavior should exist.

## :bulb: Proposed solution
This PR defaults `enableBehavior` to ALL, i.e. the question is shown only when all the `enableWhen` expressions are satisfied. We also update a UI test to cover this situation.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
@vishnuravi Thanks for the changes! The change is compact and looks good to me.

I approve the PR but It would be great if we can validate the multiple conditions in one or more unit test (probably no UI test needed if we can just inspect the ORKNavigableOrderedTask in a unit test) to validate that the enabled behavior is correctly translated for an or, and, and none-provided case we don't run into any regressions there in the future.
Would be great if you can add these unit tests before merging the PR itself. Free free to re-request a review if I should take a look at them but I am also fine with merging it after they pass.

**Comment by vishnuravi** [comment]
@PSchmiedmayer I have updated the multipleEnableWhen UI test to cover this situation, is that sufficient, or do you think it also needs a separate unit test?

**Comment by PSchmiedmayer** [comment]
@vishnuravi: I think the UI tests cover most of the parts, but it would be good if we could easily verify that the conditions are transformed and concatenated correctly, as the UI test just covers one golden path through the survey and not all possible answers choices.

You could write a unit test that inspects the skip rules, similar to how the `NavigationRulesTests` are currently written. We can extend that group of tests to work with multiple enable when clauses and inspect the resulting `ORKSkipStepNavigationRule` beyond being nil or non-nil.

**Comment by vishnuravi** [comment]
Sounds good, will address this in another PR as the UI tests should be sufficient for the moment.

**Pull request merged**
